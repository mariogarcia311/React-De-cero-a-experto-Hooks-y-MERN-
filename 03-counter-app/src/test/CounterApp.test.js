import CounterApp from "../CounterApp"
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"

describe('pruebas primera APP', () => {
    

    test('debe de mostrar CounterApp correctamente ', () => {
        const value=0;
        const wrapper= shallow(< CounterApp value={value} />)
        expect(wrapper).toMatchSnapshot();
    })
    test('debe mostrat subtítulo enviado por props', () => {
        
        const value=100;
        const wrapper= shallow(
            < CounterApp 
                value={value}
                
            />
        );
        
        const textoParrafo= wrapper.find('h2').text().trim();
        expect(textoParrafo).toBe('100')
        
    })
    test('debe incrementar con botón +1', () => {
        const value=0;
        const wrapper= shallow(< CounterApp value={value} />)

        wrapper.find('button').at(2).simulate('click');

        const counterText= wrapper.find('h2').text().trim();

        expect(counterText).toBe('1')
        
    })
    test('debe incrementar con botón -1', () => {
        const value=0;
        const wrapper= shallow(< CounterApp value={value} />)

        wrapper.find('button').at(0).simulate('click');

        const counterText= wrapper.find('h2').text().trim();

        expect(counterText).toBe('-1')
        
    })
    
})