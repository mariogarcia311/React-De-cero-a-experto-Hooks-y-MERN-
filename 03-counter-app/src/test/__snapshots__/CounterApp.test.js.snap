// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas primera APP debe de mostrar CounterApp correctamente  1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
